library/alloc/src/str.rs

```diff
@@ -661,3 +687,11 @@ fn convert_while_ascii(b: &[u8], convert: fn(&u8) -> u8) -> Vec<u8> {
 
     out
 }
+#[inline]
+#[cfg(not(test))]
+#[cfg(not(no_global_oom_handling))]
+fn replace_ascii(utf8_bytes: &[u8], from: u8, to: u8) -> String {
```

Comment:
Nit: This function will call `from_utf8_unchecked()` on _any_ `&[u8]` without checking it's precondition. Even if it's only used locally, it should be marked `unsafe` to force the caller to recognize that fact.

Reply:
> Nit: This function will call `from_utf8_unchecked()` on _any_ `&[u8]` without checking it's precondition. Even if it's only used locally, it should be marked `unsafe` to force the caller to recognize that fact.

Good point. ```replace_ascii()``` is now unsafe.